### ISSUE TITLE
unable to load azure-devops extension in Azure Function

### ISSUE THREAD

**Issue by smholvoet** [Azure/azure-cli#21134]
**Describe the bug**
Hi 👋

A current shortcoming of PowerShell based Azure Functions is the inability to natively run Azure CLI commands 👉see https://github.com/Azure/azure-functions-powershell-worker/issues/221. A [workaround](https://github.com/Azure/azure-functions-powershell-worker/issues/221#issuecomment-546039332) which is mentioned in this issue comes down to:

- deploying Azure CLI to the underlying storage account where the Function is running by coping the `CLI2` folder
- creating an alias for the `az` command in the `profile.ps1` file

I have tried this and even though I was able to verify Azure CLI was loading correctly, using `az version`, I am having trouble loading the [Azure DevOps](https://docs.microsoft.com/en-us/azure/devops/cli/?view=azure-devops) extension.

`az version`:
![image](https://user-images.githubusercontent.com/13507392/151397181-ef33aa40-ddd9-4728-95fd-462d88be144f.png)

Installing the Azure DevOps extension:
![image](https://user-images.githubusercontent.com/13507392/151397288-0ea18d14-f235-4106-a93d-d6e2cd481afa.png)

The screenshot above would suggest it loads fine, but I'm unable to call any [`az devops`](https://docs.microsoft.com/en-us/cli/azure/devops?view=azure-cli-latest) commands, which results in the following errors:

![image](https://user-images.githubusercontent.com/13507392/151397541-e5c2cacf-43cb-4ef1-a264-0859adf648f6.png)

I tried specifying the `--debug` flag, to no avail.

**To Reproduce**
- Create a [PowerShell](https://docs.microsoft.com/en-us/azure/azure-functions/create-first-function-vs-code-powershell) based Azure Function
- Deploy Azure CLI to the underlying storage account and create an alias for the `az` command (see https://github.com/Azure/azure-
- Add the following snippet to the `run.ps1` file:
- Run the Function
  ```
  az version
  az extension add --name azure-devops
  az extension list
  ```

**Expected behavior**
Azure CLI should be able to load the extension succesfully

**Environment summary**
ℹ Manually copied the `CLI2` folder (see bug description)

```
{
  "azure-cli": "2.12.0",
  "azure-cli-core": "2.12.0",
  "azure-cli-telemetry": "1.0.6",
  "extensions": {
    "azure-devops": "0.21.0",
    "interactive": "0.4.5"
  }
}
```

**Comment by yonzhan** [comment]
route to service team

**Comment by smholvoet** [comment]
I noticed my Azure CLI version was shamefully outdated, so I've tried updating my local install to the latest version. I then once again uploaded the entire `CLI2` folder to the shared folder of my Azure Function... and now everything works fine. Seems like my somewhat outdated version (2.12.0) did not get along well with whatever's running in Azure Function environments 🤷‍♂️

Lesson learned: [update](https://docs.microsoft.com/en-us/cli/azure/update-azure-cli) your Azure CLI regularly  version boys and girls!
